src/BaseControllerV2.ts

```diff
@@ -82,19 +82,33 @@ export type StateMetadata<T> = {
 };
 
 /**
- * Metadata for a single state property
- *
- * @property persist - Indicates whether this property should be persisted
- *   (`true` for persistent, `false` for transient), or is set to a function
- *   that derives the persistent state from the state.
- * @property anonymous - Indicates whether this property is already anonymous,
- *   (`true` for anonymous, `false` if it has potential to be personally
- *   identifiable), or is set to a function that returns an anonymized
- *   representation of this state.
+ * Metadata for a single state property.
  */
 export interface StatePropertyMetadata<T> {
+  /**
+   * Indicates whether this property should be persisted (`true` for persistent,
+   * `false` for transient), or is set to a function that derives the persistent
+   * state from this state.
+   */
   persist: boolean | StateDeriver<T>;
+
+  /**
+   * Indicates whether this property is already anonymous, (`true` for
+   * anonymous, `false` if it has potential to be personally identifiable),
+   * or is set to a function that returns an anonymized representation of this
+   * state.
+   */
   anonymous: boolean | StateDeriver<T>;
+
+  /**
+   * Indicates whether this property is communicated to consumers by default
+   * (`true` if so,`false` if not), or is set to a function that derives
+   * such state from this state.
```

Comment:
This seems like it could be improved :thinking: I'll take a stab at this later. It would be great to highlight that the "private" state can still end up being persisted for example.

Reply:
I've made an attempt to improve this in https://github.com/MetaMask/controllers/pull/483/commits/1a6814e50e8dbacc01b5c09815ace31db193bdde